### ISSUE TITLE
Add tolerance to has_affine_map()

### ISSUE THREAD

**Issue by loganharbour** [libMesh/libmesh#2799]
In ray tracing, in cases where a QUAD4 side is _slightly_ not affine, we can get away with tracing on a planar face even though it's not planar.

I'd like to add a tolerance to `has_affine_map()` for this purpose. Thoughts?

**Comment by jwpeterson** [comment]
Hmm... `Quad4::has_affine_map()` already uses `relative_fuzzy_equals()`, which uses a default tolerance of `TOLERANCE`, so I think it would make sense to add a `Real tol = TOLERANCE` default parameter to `has_affine_map()` and pass it through to `relative_fuzzy_equals()`.

**Comment by loganharbour** [comment]
Yeah, that's exactly what I'm thinking. Alright - will do. Thanks guys.

**Comment by loganharbour** [comment]
Also: `TOLERANCE` is a pretty loose tolerance, isn't it? Just curious - AFAIK `TOLERANCE * TOLERANCE` is used in most places for checks similar to this.

**Comment by roystgnr** [comment]
I'd be quite happy with changing the default to ```TOLERANCE * TOLERANCE```.  I don't recall putting much thought into the initial decision.  And the main effect of a too-tight tolerance would be to cause us to choose slower code paths, whereas the main effect of a too-loose tolerance would be numerical error, so I'd rather err on the side of too-tight.

**Comment by loganharbour** [comment]
> I'd be quite happy with changing the default to `TOLERANCE * TOLERANCE`.

Sounds good. Time to see what I can break.
